javascript/src/components/inputs/InputDropdown.jsx

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var React = require('react/addons');
+var InputsStore = require('../../stores/inputs/InputsStore');
+var Input = require('react-bootstrap').Input;
+
+var InputDropdown = React.createClass({
+    mixins: [React.addons.LinkedStateMixin],
+    PLACEHOLDER: "placeholder",
+    getInitialState() {
+        return {
+            inputs: [],
+            selectedInput: this.PLACEHOLDER
+        };
+    },
+    loadData() {
+        InputsStore.list((inputs) => {
+            this.setState({inputs: inputs});
+        });
+    },
+    componentDidMount() {
+        if (!this.props.inputs) {
```

Comment:
This still doesn't work for three reasons:
- The parent defines the inputs on the `getInitialState` as undefined, and when this method is reached the actual inputs are still not loaded, so the if doesn't work as expected
- The parent stores inputs as a `Map`, but this component does it as a JS array
- We should also update the inputs when the parent changes them, so we need to add a `componentWillUpdateProps` and update the state accordingly